site/content/docs/Advanced/Ransomware Protection/_index.md

```diff
@@ -126,11 +123,20 @@ When this is configured, the retention mode can be set to either compliance or g
 
 Follow [these steps](https://learn.microsoft.com/en-us/azure/storage/blobs/versioning-enable) to enable versioning on the storage account and [these steps](https://learn.microsoft.com/en-us/azure/storage/blobs/immutable-policy-configure-version-scope) to enable version-level immutability support on the container or related storage account.
 
-On Kopia side `--retention-mode COMPLIANCE --retention-period <retention time>` should be set like above. 
+On Kopia side `--retention-mode COMPLIANCE --retention-period <retention time>` should be set like above.
+
+To have continuous protection it is also necessary to run: `kopia maintenance set --extend-object-locks true`
+* Note that the `full-interval` must be at least 1 day shorter than the `retention-period` or Kopia will not allow you to enable Object Lock extension
 
 ### Google protection
 
 Kopia supports ransomware protection for Google in a similar manner to S3. The bucket must have both versioning and object retention enabled.
 When this is configured, the retention mode can be set to either compliance or governance mode. In both cases the blobs will be in [Locked](https://cloud.google.com/storage/docs/object-lock#overview) mode.
 
 On Kopia side `--retention-mode COMPLIANCE --retention-period <retention time>` should be set like above.
+
+To have continuous protection it is also necessary to run: `kopia maintenance set --extend-object-locks true`
+* Note that the `full-interval` must be at least 1 day shorter than the `retention-period` or Kopia will not allow you to enable Object Lock extension
+
+If using minimal permissions with the credentials,
+`storage.objects.setRetention` permission is also required.
```

Comment:
@KastenMike Can you add the set of minimal permissions required? Thanks.

Reply:
@KastenMike Can you look for a good place in the docs to include these? Maybe in the (GCS) section for the storage providers.

Perhaps the permissions can be divided into
- required for read-only ops
  ```
  storage.buckets.get
  storage.objects.get
  storage.objects.list
  ```
- (additional perms) required for read-write ops
  ```
  storage.objects.update
  storage.objects.create
  storage.objects.delete
  ```
- (additional perms)required for immutability / ransomware protection
  ```
  storage.objects.setRetention
  ```

Reply:
https://github.com/kopia/kopia/pull/4193